### PR TITLE
Integrating Coverage.py with cesium

### PR DESCRIPTION
**Pull request by arkwave**
1) updated travis.yml to run codecov after build passes.
2) added in a .coveragerc file to stop `coverage.py` from testing scikit-learn imports
3) updated requirements.txt to include coverage.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -93,4 +93,6 @@ install:
 
 script: .drone/travis_script.sh
 
-after_success: .drone/deploy_docs.sh
+after_success: 
+  - .drone/deploy_docs.sh
+  - codecov
```
Comment (stefanv) [inline .travis.yml]: Look at deploy_docs.sh to see how to only run this on one Python version

**Comment by stefanv** [comment]
Problem: the test suite will now get run twice, once without and once with coverage, under Python 3.5.  Can we change it so that it only is run once?

File: .drone/travis_script.sh [hunk 2]
```diff
@@ -7,7 +7,17 @@ source ~/envs/cesium/bin/activate
 
 section "Tests"
 
-make ${TEST_TARGET}
+if [[$COVERAGE == 1]]
+
```
Comment (stefanv) [inline .drone/travis_script.sh]: Remove blank line

File: .drone/travis_script.sh [hunk 3]
```diff
@@ -7,7 +7,17 @@ source ~/envs/cesium/bin/activate
...
+then
+    echo "Python version: 3.4/3.5"
```
Comment (stefanv) [inline .drone/travis_script.sh]: Remove this line

File: .drone/travis_script.sh [hunk 4]
```diff
@@ -7,7 +7,17 @@ source ~/envs/cesium/bin/activate
...
+    echo "Running coverage.py"
+    nosetests -v --exe --with-coverage
+else
+    echo "Python version: 2.x"
+    echo "Coverage not set up for python 2.x"
+    make ${TEST_TARGET}
```
Comment (stefanv) [inline .drone/travis_script.sh]: It looks like TEST_TARGET is unused.  Let's remove it from .travis.yml, and then do something like this:

```
if [[ $COVERAGE == 1 ]]; then
    NOSE_FLAGS="--with-coverage"
fi

nose -v $NOSE_FLAGS cesium
```

**Comment by stefanv** [comment]
We probably also need a command to upload the coverage report to cov.io?

**Comment by arkwave** [comment]
@stefanv I'm not entirely sure, this seems to suggest that integration is handled automatically: https://github.com/codecov/codecov-python (specifically the Configuration section). Thoughts?

**Comment by stefanv** [comment]
It looks like they have the following:

```
after_success:
  - codecov
```

**Comment by arkwave** [comment]
Hmm I am a little confused in that case. Initially, that same line `after_success: codecov` was marked for deletion because it led to the test suite running twice. However, in the `scikit-image` repo, additional dependencies are tested with coverage in the same way as they're done here in `travis_script.sh`, but `after succcess: codecov` is still included in the `travis.yml` file. Wouldn't this also lead to coverage running twice, or is there something I'm misunderstanding?

**Comment by stefanv** [comment]
In most cases, that command just won't do anything, since there's no
coverage information to upload:
https://travis-ci.org/scikit-image/scikit-image/jobs/174340273#L7563

**Comment by arkwave** [comment]
Alright, I have added that line back in. Thanks for catching it. Also, does the cesium-ml repo might have to be registered at `codecov.io` for coverage reports to be published? Or is that more to centralize the coverage reports at a single location?

**Comment by stefanv** [comment]
I've registered cesium with them now, so we should be good to go!

**Comment by stefanv** [comment]
Looks like codecov is missing (no pip install?):
https://travis-ci.org/cesium-ml/cesium/jobs/175605412#L1617

Also, code coverage is enabled for two items in the matrix, instead
of just one.

**Comment by stefanv** [comment]
Excellent, thanks @arkwave !

**Comment by bnaul** [comment]
This does not appear to actually work (see: all the PRs since this was merged).

**Comment by stefanv** [comment]
See https://github.com/cesium-ml/cesium/pull/237

**Pull request merged**